engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -210,31 +245,32 @@ public BlockRegion minZ(int z) {
     }
 
     /**
-     * Get the block coordinate minimum corner.
+     * Get the block coordinate of the minimum corner.
      *
      * @param dest will hold the result
+     * @return {@code dest} after the result has been set
      */
     public Vector3i getMin(Vector3i dest) {
         return dest.set(minX, minY, minZ);
     }
 
     /**
-     * Sets the minimum coordinate of the first block for <code>this</code> {@link BlockRegion}
+     * Set the coordinates of the minimum corner for this region.
      *
-     * @param min the first coordinate of the first block
-     * @return this
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if any dimension is greater than the respective component of the max
+     *         corner
      */
     public BlockRegion setMin(Vector3ic min) {
         return this.setMin(min.x(), min.y(), min.z());
     }
 
     /**
-     * sets the minimum block for this {@link BlockRegion}
+     * Set the coordinates of the minimum corner for this region.
      *
-     * @param minX the x coordinate of the first block
-     * @param minY the y coordinate of the first block
-     * @param minZ the z coordinate of the first block
-     * @return this
+     * @return this region (after modification)
+     * @throws IllegalArgumentException if any dimension is greater than the respective component of the max
+     *         corner
      */
     public BlockRegion setMin(int minX, int minY, int minZ) {
```

Comment:
I just realized this precondition might be a problem if you setMin above max and then aftewards you setmax. The final region is going to be valid but you're going to hit this exception. Its an extra burden to require the min and max to be set once.

Reply:
Well, I thought we agreed with @4Denthusiast that we are trying for a design where a BlockRegion is (almost always) valid - which means that you have to pay attention to details like this. In this case you know that min and max will work out in the end, but what happens if the calls are split over several methods or even different systems handling it?

We already have `set(BlockRegion)`, maybe we also need `set(minX, minY, minZ, maxX, maxY, maxZ)` which would solve your problem, I guess?